crates/polars-lazy/src/physical_plan/executors/filter.rs

```diff
@@ -35,6 +35,9 @@ impl Executor for FilterExec {
             state.insert_has_window_function_flag()
         }
         let s = self.predicate.evaluate(&df, state)?;
+        if state.cache_window() {
```

Comment:
Might replace this by `if has_window` to reduce the overhead of atomic load? 🤔

Reply:
Yes an you put the `cache_window` and the `clear_window_expr` behind a `if self.has_windows`?

Reply:
Do you mean to replace it with

```rust
if state.has_window() && state.cache_window() {
   state.clear_window_expr_cache()
}
```

Reply:
This is what we do during projection.

```rust
    if has_windows {
        state.clear_window_expr_cache();
    }
```

Reply:
I see, removed it.